src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/PaymentProcess/SendMetricsReportEmailCommandHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using NServiceBus;
+using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Data.Reports;
+using SFA.DAS.EmployerIncentives.Data.Reports.Metrics;
+using SFA.DAS.EmployerIncentives.Infrastructure.Configuration;
+using SFA.DAS.EmployerIncentives.Reports;
+using SFA.DAS.Notifications.Messages.Commands;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.PaymentProcess
+{
+    public class SendMetricsReportEmailCommandHandler : ICommandHandler<SendMetricsReportEmailCommand>
+    {
+        private readonly IReportsDataRepository _reportsDataRepository;
+        private readonly IReportsRepository _reportsRepository;
+        private readonly IConfiguration _configuration;
+        private readonly ICommandPublisher _commandPublisher;
+        private readonly EmailTemplateSettings _emailTemplates;
+        
+        private const string CollectionPeriodToken = "periodName";
+        private const string AcademicYearToken = "academicYear";
+        private const string AmountToBePaidToken = "amountToBePaid";
+        private const string AmountFailingValidationToken = "amountFailingValidation";
+        private const string MetricsReportDownloadToken = "metricsReportDownloadLink";
+
+
+        public SendMetricsReportEmailCommandHandler(IReportsDataRepository reportsDataRepository,
+                                                    IReportsRepository reportsRepository,
+                                                    ICommandPublisher commandPublisher,
+                                                    IOptions<EmailTemplateSettings> emailTemplates,
+                                                    IConfiguration configuration)
+        {
+            _reportsDataRepository = reportsDataRepository;
+            _reportsRepository = reportsRepository;
+            _commandPublisher = commandPublisher;
+            _emailTemplates = emailTemplates.Value;
+            _configuration = configuration;
+        }
+
+        public async Task Handle(SendMetricsReportEmailCommand command, CancellationToken cancellationToken = default)
+        {
+            var report = await _reportsDataRepository.Execute<MetricsReport>();
+
+            var templateId = _emailTemplates.MetricsReport.TemplateId;
+
+            var personalisationTokens = new Dictionary<string, string>
+            {
+                { CollectionPeriodToken, command.CollectionPeriod.PeriodNumber.ToString() },
```

Comment:
The period and year are on the MetricsReport. .  Please add a guard clause to throw an error id the command and report values for AcademicYear and Period differ.

Reply:
Fixed